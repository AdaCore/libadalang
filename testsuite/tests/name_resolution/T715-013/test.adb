procedure Test is
   package P is
      type T;
      type T_CW is access all T'Class;

      function Foo (X : access T) return Boolean is (True);

      type T is tagged null record;
   end P;

   package Q is
      type U;
      type U_CW is access all U'Class;

      function Foo (X : access U) return Integer is (2);

      type U is tagged null record;
   end Q;

   package R is
      type T is new P.T with null record;
      type U is new Q.U with null record;
   end R;

   use R;

   X : access U;
begin
   begin
      case Foo (X) is
         when 1 => null;
         when others => null;
      end case;
   end;
   pragma Test_Block;
end Test;
