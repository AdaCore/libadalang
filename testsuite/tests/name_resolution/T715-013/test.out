Analyzing test.adb
##################

Resolving xrefs for node <CaseStmt test.adb:30:7-33:16>
*******************************************************

Expr: <CallExpr test.adb:30:12-30:19>
  references: <DefiningName test.adb:15:16-15:19>
  type:       <TypeDecl ["Integer"] __standard:4:3-4:54>
Expr: <Id "Foo" test.adb:30:12-30:15>
  references: <DefiningName test.adb:15:16-15:19>
  type:       <TypeDecl ["Integer"] __standard:4:3-4:54>
Expr: <Id "X" test.adb:30:17-30:18>
  references: <DefiningName test.adb:27:4-27:5>
  type:       <AnonymousTypeDecl ["None"] test.adb:27:8-27:16>

Resolving xrefs for node <CaseStmtAlternative test.adb:31:10-31:25>
*******************************************************************

Expr: <Int test.adb:31:15-31:16>
  references: None
  type:       <TypeDecl ["Integer"] __standard:4:3-4:54>

Resolving xrefs for node <NullStmt test.adb:31:20-31:25>
********************************************************


Resolving xrefs for node <CaseStmtAlternative test.adb:32:10-32:30>
*******************************************************************


Resolving xrefs for node <NullStmt test.adb:32:25-32:30>
********************************************************



Done.
